File: sdkwork-app-sdk-java/src/main/java/com/sdkwork/app/api/ShareApi.java
package com.sdkwork.app.api;

import com.sdkwork.app.http.HttpClient;
import com.sdkwork.app.model.*;
import java.util.List;
import java.util.Map;

public class ShareApi {
    private final HttpClient client;
    
    public ShareApi(HttpClient client) {
        this.client = client;
    }

    /** 更新分享设置 */
    public PlusApiResultShareRecordVO updateShareSettings(String shareId, ShareUpdateForm body) throws Exception {
        return (PlusApiResultShareRecordVO) client.put(ApiPaths.appPath("/share/" + shareId + ""), body);
    }

    /** 取消分享 */
    public PlusApiResultVoid cancel(String shareId) throws Exception {
        return (PlusApiResultVoid) client.delete(ApiPaths.appPath("/share/" + shareId + ""));
    }

    /** 创建分享 */
    public PlusApiResultShareCreateVO createShare(ShareCreateForm body) throws Exception {
        return (PlusApiResultShareCreateVO) client.post(ApiPaths.appPath("/share"), body);
    }

    /** 访问分享 */
    public PlusApiResultVoid visit(String shareCode, ShareVisitForm body) throws Exception {
        return (PlusApiResultVoid) client.post(ApiPaths.appPath("/share/" + shareCode + "/visit"), body);
    }

    /** 验证分享密码 */
    public PlusApiResultShareVerifyVO verifySharePassword(String shareCode, ShareVerifyForm body) throws Exception {
        return (PlusApiResultShareVerifyVO) client.post(ApiPaths.appPath("/share/" + shareCode + "/verify"), body);
    }

    /** 上报分享 */
    public PlusApiResultVoid track(ShareTrackForm body) throws Exception {
        return (PlusApiResultVoid) client.post(ApiPaths.appPath("/share/track"), body);
    }

    /** 生成分享海报 */
    public PlusApiResultSharePosterVO generateSharePoster(SharePosterForm body) throws Exception {
        return (PlusApiResultSharePosterVO) client.post(ApiPaths.appPath("/share/poster"), body);
    }

    /** 领取邀请奖励 */
    public PlusApiResultInviteRewardClaimVO claimInviteReward(String rewardId) throws Exception {
        return (PlusApiResultInviteRewardClaimVO) client.post(ApiPaths.appPath("/share/invite/rewards/" + rewardId + "/claim"), null);
    }

    /** 生成邀请链接 */
    public PlusApiResultInviteLinkVO generateInviteLink(InviteLinkForm body) throws Exception {
        return (PlusApiResultInviteLinkVO) client.post(ApiPaths.appPath("/share/invite/link"), body);
    }

    /** 获取访问记录 */
    public PlusApiResultPageShareVisitorVO getShareVisitors(String shareId, Map<String, Object> params) throws Exception {
        return (PlusApiResultPageShareVisitorVO) client.get(ApiPaths.appPath("/share/" + shareId + "/visitors"), params);
    }

    /** 获取分享统计 */
    public PlusApiResultShareStatisticsVO getShareStatistics(String shareId) throws Exception {
        return (PlusApiResultShareStatisticsVO) client.get(ApiPaths.appPath("/share/" + shareId + "/statistics"));
    }

    /** 获取分享信息 */
    public PlusApiResultShareInfoVO getShareInfo(String shareCode, Map<String, Object> params) throws Exception {
        return (PlusApiResultShareInfoVO) client.get(ApiPaths.appPath("/share/" + shareCode + ""), params);
    }

    /** 获取分享平台配置 */
    public PlusApiResultListSharePlatformVO getSharePlatforms() throws Exception {
        return (PlusApiResultListSharePlatformVO) client.get(ApiPaths.appPath("/share/platforms"));
    }

    /** 获取我的分享 */
    public PlusApiResultPageShareRecordVO listMyShares(Map<String, Object> params) throws Exception {
        return (PlusApiResultPageShareRecordVO) client.get(ApiPaths.appPath("/share/my-shares"), params);
    }

    /** 获取邀请记录 */
    public PlusApiResultPageInviteRecordVO getInviteRecords(Map<String, Object> params) throws Exception {
        return (PlusApiResultPageInviteRecordVO) client.get(ApiPaths.appPath("/share/invite/records"), params);
    }

    /** 获取邀请信息 */
    public PlusApiResultInviteInfoVO getInviteInfo() throws Exception {
        return (PlusApiResultInviteInfoVO) client.get(ApiPaths.appPath("/share/invite/info"));
    }

    /** 批量取消分享 */
    public PlusApiResultVoid batchCancelShares() throws Exception {
        return (PlusApiResultVoid) client.delete(ApiPaths.appPath("/share/batch"));
    }
}

File: sdkwork-app-sdk-kotlin/src/main/kotlin/com/sdkwork/app/api/ShareApi.kt
package com.sdkwork.app.api

import com.sdkwork.app.*
import com.sdkwork.app.http.HttpClient

class ShareApi(private val client: HttpClient) {

    /** 更新分享设置 */
    suspend fun updateShareSettings(shareId: String, body: ShareUpdateForm): PlusApiResultShareRecordVO? {
        return client.put(ApiPaths.appPath("/share/$shareId"), body) as? PlusApiResultShareRecordVO
    }

    /** 取消分享 */
    suspend fun cancel(shareId: String): PlusApiResultVoid? {
        return client.delete(ApiPaths.appPath("/share/$shareId")) as? PlusApiResultVoid
    }

    /** 创建分享 */
    suspend fun createShare(body: ShareCreateForm): PlusApiResultShareCreateVO? {
        return client.post(ApiPaths.appPath("/share"), body) as? PlusApiResultShareCreateVO
    }

    /** 访问分享 */
    suspend fun visit(shareCode: String, body: ShareVisitForm): PlusApiResultVoid? {
        return client.post(ApiPaths.appPath("/share/$shareCode/visit"), body) as? PlusApiResultVoid
    }

    /** 验证分享密码 */
    suspend fun verifySharePassword(shareCode: String, body: ShareVerifyForm): PlusApiResultShareVerifyVO? {
        return client.post(ApiPaths.appPath("/share/$shareCode/verify"), body) as? PlusApiResultShareVerifyVO
    }

    /** 上报分享 */
    suspend fun track(body: ShareTrackForm): PlusApiResultVoid? {
        return client.post(ApiPaths.appPath("/share/track"), body) as? PlusApiResultVoid
    }

    /** 生成分享海报 */
    suspend fun generateSharePoster(body: SharePosterForm): PlusApiResultSharePosterVO? {
        return client.post(ApiPaths.appPath("/share/poster"), body) as? PlusApiResultSharePosterVO
    }

    /** 领取邀请奖励 */
    suspend fun claimInviteReward(rewardId: String): PlusApiResultInviteRewardClaimVO? {
        return client.post(ApiPaths.appPath("/share/invite/rewards/$rewardId/claim"), null) as? PlusApiResultInviteRewardClaimVO
    }

    /** 生成邀请链接 */
    suspend fun generateInviteLink(body: InviteLinkForm): PlusApiResultInviteLinkVO? {
        return client.post(ApiPaths.appPath("/share/invite/link"), body) as? PlusApiResultInviteLinkVO
    }

    /** 获取访问记录 */
    suspend fun getShareVisitors(shareId: String, params: Map<String, Any>? = null): PlusApiResultPageShareVisitorVO? {
        return client.get(ApiPaths.appPath("/share/$shareId/visitors"), params) as? PlusApiResultPageShareVisitorVO
    }

    /** 获取分享统计 */
    suspend fun getShareStatistics(shareId: String): PlusApiResultShareStatisticsVO? {
        return client.get(ApiPaths.appPath("/share/$shareId/statistics")) as? PlusApiResultShareStatisticsVO
    }

    /** 获取分享信息 */
    suspend fun getShareInfo(shareCode: String, params: Map<String, Any>? = null): PlusApiResultShareInfoVO? {
        return client.get(ApiPaths.appPath("/share/$shareCode"), params) as? PlusApiResultShareInfoVO
    }

    /** 获取分享平台配置 */
    suspend fun getSharePlatforms(): PlusApiResultListSharePlatformVO? {
        return client.get(ApiPaths.appPath("/share/platforms")) as? PlusApiResultListSharePlatformVO
    }

    /** 获取我的分享 */
    suspend fun listMyShares(params: Map<String, Any>? = null): PlusApiResultPageShareRecordVO? {
        return client.get(ApiPaths.appPath("/share/my-shares"), params) as? PlusApiResultPageShareRecordVO
    }

    /** 获取邀请记录 */
    suspend fun getInviteRecords(params: Map<String, Any>? = null): PlusApiResultPageInviteRecordVO? {
        return client.get(ApiPaths.appPath("/share/invite/records"), params) as? PlusApiResultPageInviteRecordVO
    }

    /** 获取邀请信息 */
    suspend fun getInviteInfo(): PlusApiResultInviteInfoVO? {
        return client.get(ApiPaths.appPath("/share/invite/info")) as? PlusApiResultInviteInfoVO
    }

    /** 批量取消分享 */
    suspend fun batchCancelShares(): PlusApiResultVoid? {
        return client.delete(ApiPaths.appPath("/share/batch")) as? PlusApiResultVoid
    }
}
